### ISSUE TITLE
repetitions in rnn architecture

### ISSUE THREAD

**Issue by sajastu** [pytorch/fairseq#611]
Hi!

I trained fairseq-lstm over my data with the following arguments:

`CUDA_VISIBLE_DEVICES=0 fairseq-train data-bin/sample_data -a lstm --dropout 0.5 --encoder-embed-dim 100 --decoder-embed-dim 100 --encoder-embed-path ~/data/glove_dir/radglove.800M.100d.txt  --decoder-embed-path ~/data/glove_dir/radglove.800M.100d.txt --save-dir checkpoints/lstm-embed`

Then in prediction time:
```
CUDA_VISIBLE_DEVICES=0 fairseq-generate data-bin/sample_data \
    --path checkpoints/lstm-embed/checkpoint_best.pt \
    --batch-size 25 --beam 5 --remove-bpe > text-lstm.txt
```

Now when looking at text-lstm.txt I see some cases like these below:

>S-2722  intraoperative fluoroscopy provided. no fluoroscopic captures were available the time of dictation. fluoroscopic time : 0 minutes 10 seconds.
 T-2722  intraoperative fluoroscopy. please refer to operative report for operative details and findings.
H-2722  -1.6138560771942139     1. no evidence of deep thrombosis of the right lower lobe with the right lower lobe with the right lower lobe with the right lower lobe lobe is not be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be be

I also used Transformer, but it went well without having these kinds of problems. Is there any argument that I need to set before doing training/evaluation?

Thanks!

**Comment by huihuifan** [comment]
I cannot see anything wrong with your parameters, but please see if the LSTM has reached as good results in perplexity as the transformer, compare the model quality and capacity. Repetition can be a sign of a poorly trained model.
